lib/services/elasticsearch.js

```diff
@@ -631,43 +632,57 @@ class ElasticSearch extends Service {
     }
 
     esRequest.body = request.input.body.bulkData;
+    const kuzzleMetaCreated = {
+      author: userId,
+      createdAt: dateNow,
+      updatedAt: null,
+      updater: null,
+      active: true,
+      deletedAt: null
+    };
+    const kuzzleMetaUpdated = {
+      updater: userId,
+      updatedAt: dateNow
+    };
 
-    // set missing index & type if possible
-    esRequest.body.forEach(item => {
+    // set missing index & type if possible and add metadata
+    let lastAction; // NOSONAR
+    let i; // NOSONAR
+    for(i = 0; i < esRequest.body.length; i++) {
+      const item = esRequest.body[i];
```

Comment:
nitpicking: I would rather use `for .. of` rather than `for` (which would avoid a // NOSONAR)

Reply:
I use `for(;;)` because it's much efficient than `for(const .. of ..)`.  
Since a bulk import can potentially contain 100K documents (200K elements in the body) any performance gain is good to take.  

Some benchmarks:  
```
aschen@gman perso/benchmark » nvm exec 6.13.1 node for_loop.js
Running node v6.13.1 (npm v3.10.10)
forIterator x 193,214 ops/sec ±0.99% (91 runs sampled)
forLoop x 3,024,144 ops/sec ±0.95% (91 runs sampled)
```
```
aschen@gman perso/benchmark » nvm exec 8.10.0 node for_loop.js
Running node v8.10.0 (npm v5.6.0)
forIterator x 4,673,407 ops/sec ±1.53% (91 runs sampled)
forLoop x 5,243,791 ops/sec ±1.05% (91 runs sampled)
```
```
function forIterator(array) {
  for (const item of array) {
    item.age = item.age * 2;
  }
}

function forLoop(array) {
  for (let i = 0; i < array.length; ++i) {
    array[i].age = array[i].age * 2;
  }
}
```

Reply:
OK, it could be fine to explain that with a 2-line comment above the loop.
(example here: https://github.com/kuzzleio/kuzzle/blob/master/lib/api/core/notifier.js#L248 )